src/sqlfluff/dialects/dialect_sparksql.py

```diff
@@ -2481,3 +2481,40 @@ class GeneratedColumnDefinitionSegment(BaseSegment):
             Ref("ColumnConstraintSegment", optional=True),
         ),
     )
+
+
+class MergeUpdateClauseSegment(ansi.MergeUpdateClauseSegment):
+    """`UPDATE` clause within the `MERGE` statement."""
+
+    type = "merge_update_clause"
+    match_grammar: Matchable = Sequence(
+        "UPDATE",
+        OneOf(
+            Sequence("SET", Ref("WildcardIdentifierSegment")),
+            Sequence(
+                Indent,
+                Ref("SetClauseListSegment"),
+                Dedent,
+            ),
+        ),
+    )
+
+
+class MergeInsertClauseSegment(ansi.MergeInsertClauseSegment):
+    """`INSERT` clause within the `MERGE` statement."""
+
+    type = "merge_insert_clause"
+    match_grammar: Matchable = Sequence(
+        "INSERT",
+        OneOf(
+            Ref("WildcardIdentifierSegment"),
+            Sequence(
+                Indent,
+                Ref("BracketedColumnReferenceListGrammar", optional=True),
```

Comment:
I don't really know why this would be optinal (just copied class from ansi.MergeInsertClauseSegment)

Reply:
I agree that it shouldn't be optional. I ran a test and got the following error when it was removed `Error in SQL statement: ParseException`

Reply:
You got that error in Sparksql? Or in SQLFluff?

If the former, then let's remove it.

```suggestion
                Ref("BracketedColumnReferenceListGrammar"),
```

Reply:
The former, in SparkSQL